File: Arrays and Hashing/169-Majority-Element/169.cs
public class Solution {
    public int MajorityElement(int[] nums) {
        int count = 0;
        int candidate = 0;
        foreach(int num in nums){
            if(count == 0){
                candidate = num;
            }
            if(candidate == num){
                count++;
            }
            else{
                count--;
            }
        }
        return candidate;
    }
}

File: Arrays and Hashing/169-Majority-Element/169.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
    let count = 0;
    let candidate = 0;
    for(let num of nums){
        if(count == 0){
            candidate = num;
        }
        if(candidate == num){
            count++;
        }
        else{
            count--;
        }
    }
    return candidate;
};